src/test/java/api/support/APITests.java

```diff
@@ -251,105 +257,47 @@
   protected final FeeFineAccountFixture feeFineAccountFixture = new FeeFineAccountFixture();
 
   protected APITests() {
-    this(true);
+    this(true, false);
   }
 
-  protected APITests(boolean initialiseCirculationRules) {
+  protected APITests(boolean initialiseCirculationRules, boolean enableLoanHistory) {
     this.initialiseCirculationRules = initialiseCirculationRules;
+    setLoanHistoryEnabled(enableLoanHistory);
   }
 
   @BeforeClass
   public static void beforeAll() throws InterruptedException, ExecutionException,
     TimeoutException {
 
-    deployVerticles();
+    if (okapiAlreadyDeployed) {
+      return;
+    }
 
-    //Delete everything first just in case
-    deleteAllRecords();
+    deployVerticles();
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+      try {
+        undeployVerticles();
+      } catch (Exception ex) {
+        Assert.fail("Failed to undeploy verticle: " + ex);
+      }
+    }));
+    okapiAlreadyDeployed = true;
```

Comment:
What issue did you discover that required this code?

Reply:
See comment https://github.com/folio-org/mod-circulation/pull/561#discussion_r431793236